openstack/client.go

```diff
@@ -261,7 +262,19 @@ func NewBlockStorageV2(client *gophercloud.ProviderClient, eo gophercloud.Endpoi
 
 // NewSharedFileSystemV2 creates a ServiceClient that may be used to access the v2 shared file system service.
 func NewSharedFileSystemV2(client *gophercloud.ProviderClient, eo gophercloud.EndpointOpts) (*gophercloud.ServiceClient, error) {
-	return initClientOpts(client, eo, "sharev2")
+	sc, err := initClientOpts(client, eo, "sharev2")
+	if err != nil {
+		return sc, err
+	}
+
+	apiVersion, err := sfsapi.Get(sc, "v2").Extract()
+	if err != nil {
+		return sc, err
+	}
+
+	sc.Microversion = apiVersion.Version
```

Comment:
The `Microversion` in `ServiceClient` is the version the user wants to use. Why set it here to the maximum version allowed by the OpenStack deployment?